models/tts/valle/valle_trainer.py

```diff
@@ -264,3 +266,47 @@ def add_arguments(parser: argparse.ArgumentParser):
                 default="1",
                 help="0: train all modules, 1: AR Decoder, 2: NAR Decoder",
             )
+
+    def _build_dataloader(self):
+        if not self.cfg.train.use_dynamic_batchsize:
+            return super()._build_dataloader()
+        Dataset, Collator = self._build_dataset()
+        train_dataset = Dataset(self.cfg, self.cfg.dataset[0], is_valid=False)
```

Comment:
Does it only work on dataset[0] instead of all elements of the dataset list?

Reply:
cfg.dataset specifics a dataset list to be processed, but your code seems to only process the first dataset (self.cfg.dataset[0]). How about other datasets?

Reply:
done